File: pushpop/stack.py
max=20
top=0
array=[None]*max

def push(item):
    global top, array
    if top>20:
        print("Overflow!")
        return
    array[top]=item
    top+=1
    print("Item pushed!")
    
def pop():
    global top, array
    if top<0:
        print("Underflow!")
        return
    del array[-1]
    top-=1
    print("Item popped!")

def printstack():
    global top, array
    for i in range(top):
        print(array[i],end=", ")

def menu():
    while True:
        print("\n1. push\n2. pop\n3. print\n4. exit")
        opt=int(input("Enter option: "))
        if opt==1:
            item=int(input("Enter the number: "))
            push(item)
        elif opt==2:
            pop()
        elif opt==3:
            printstack()
        elif opt==4:
            return
        else:
            print("Please enter a valid choice!")


def stack():
    menu()

if __name__=="__main__":
    stack()

File: stack.java
class Stack {
	int limit = 100;
	int top;
	int a[] = new int[limit];

    Stack()
	{
		top = -1;
	}

	boolean isEmpty()
	{
		if (top==-1)
        return true;
        else
        return false;
	}
	

	boolean push(int value)
	{
		if (top >= (limit - 1)) {
			System.out.println("Stack Overflow");
			return false;
		}
		else {
			a[++top] = value;
			System.out.println(value + " pushed into stack");
			return true;
		}
	}

	int pop()
	{
		if (top < 0) {
			System.out.println("Stack Underflow");
			return 0;
		}
		else {
			int value = a[top--];
			return value;
		}
	}

	int peek()
	{
		if (top < 0) {
			System.out.println("Stack Underflow");
			return 0;
		}
		else {
			int value = a[top];
			return value;
		}
	}
}
class Main {
	public static void main(String args[])
	{
		Stack s = new Stack();
		s.push(1);
		s.push(2);
		s.push(3);
        s.push(4);
		System.out.println(s.pop() + " Popped from stack");
        System.out.println(s.peek() + " Peeked from Stack");
        System.out.println("Stack is Empty :"+s.isEmpty());
	}
}
